tests/lib/rules/newline-before-return.js

```diff
@@ -112,77 +112,90 @@ ruleTester.run("newline-before-return", rule, {
+            errors: ["Expected newline before return statement."],
+            output: "function a() {\nfor (b in c) {\nd();\n\nreturn;\n}\n}"
         },
         {
             code: "function a() {\nfor (b of c) {\nd();\nreturn;\n}\n}",
             parserOptions: { ecmaVersion: 6 },
-            errors: ["Expected newline before return statement."]
+            errors: ["Expected newline before return statement."],
+            output: "function a() {\nfor (b of c) {\nd();\n\nreturn;\n}\n}"
         },
         {
             code: "function a() {\nif (b) {\nc();\n}\n//comment\nreturn b;\n}",
-            errors: ["Expected newline before return statement."]
-        },
-        {
-            code: "function a() {\nif (b) {\nc();\n}\n//comment\nreturn b;\n}",
-            errors: ["Expected newline before return statement."]
+            errors: ["Expected newline before return statement."],
+            output: "function a() {\nif (b) {\nc();\n}\n//comment\nreturn b;\n}"
         },
         {
             code: "function a() {\n/*comment\ncomment*/\nif (b) {\nc();\nreturn b;\n} else {\n//comment\n\nreturn d;\n}\n/*multi-line\ncomment*/\nreturn e;\n}",
-            errors: ["Expected newline before return statement.", "Expected newline before return statement."]
+            errors: ["Expected newline before return statement.", "Expected newline before return statement."],
+            output: "function a() {\n/*comment\ncomment*/\nif (b) {\nc();\n\nreturn b;\n} else {\n//comment\n\nreturn d;\n}\n/*multi-line\ncomment*/\nreturn e;\n}"
         },
         {
             code: "function a() {\nif (b) { return; } //comment\nreturn c;\n}",
-            errors: ["Expected newline before return statement."]
+            errors: ["Expected newline before return statement."],
+            output: "function a() {\nif (b) { return; } //comment\n\nreturn c;\n}"
         },
         {
             code: "function a() {\nif (b) { return; } /*multi-line\ncomment*/\nreturn c;\n}",
-            errors: ["Expected newline before return statement."]
+            errors: ["Expected newline before return statement."],
+            output: "function a() {\nif (b) { return; } /*multi-line\ncomment*/\nreturn c;\n}"
         },
         {
             code: "function a() {\nif (b) { return; }\n/*multi-line\ncomment*/ return c;\n}",
-            errors: ["Expected newline before return statement."]
+            errors: ["Expected newline before return statement."],
+            output: "function a() {\nif (b) { return; }\n/*multi-line\ncomment*/ return c;\n}"
         },
         {
             code: "function a() {\nif (b) { return; } /*multi-line\ncomment*/ return c;\n}",
-            errors: ["Expected newline before return statement."]
+            errors: ["Expected newline before return statement."],
+            output: "function a() {\nif (b) { return; } /*multi-line\ncomment*/ \nreturn c;\n}"
```

Comment:
If so, could we do a check like this?

```js
function canFix(node) {
   const lastLeadingComment = node.leadingComments ? node.leadingComments[node.leadingComments.length - 1] : null;
   const tokenBefore = sourceCode.getTokenBefore(node);

   if (!lastLeadingComment || (lastLeadingComment && lastLeadingComment.loc.end.line === tokenBefore.loc.end.line)) {
       return false;
   }

   return true;
}
```
Haven't actually tried the above - so might not quite work - but you get the idea! I don't think we have to rely on the `calcCommentLines` check (since that check is designed to just ensure there's a blank newline somewhere between the return statement and the previous token) and use `leadingComment` locations to help us determine if we can fix or not.